File: 백준/Silver/2217. 로프/로프.py
import sys

input = sys.stdin.readline

n = int(input())

weights = []

for _ in range(n):
    weights.append(int(input()))

weights.sort()
max_total_weight = 0

for i, w in enumerate(weights):
    curr = w * (len(weights) - i)
    max_total_weight = max(curr, max_total_weight)

print(max_total_weight)

File: 백준/Silver/2217. 로프/로프.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;

public class Main {

    public static void main(String[] args) throws IOException {
        try(BufferedReader reader = new BufferedReader(new InputStreamReader(System.in))) {
            int n = Integer.parseInt(reader.readLine());
            int[] weights = new int[n];

            for (int i = 0; i < n; i++) {
                weights[i] = Integer.parseInt(reader.readLine());
            }

            System.out.println(solution(n, weights));
        }
    }

    private static int solution(int n, int[] weights) {
        Arrays.sort(weights);

        int max = 0;
        for (int i = 0; i < n; i++) {
            int weight = weights[i] * (n - i);
            max = Math.max(max, weight);
        }

        return max;
    }
}
